csharp/extractor/Semmle.Extraction.CSharp/Entities/Compilations/Compilation.cs

```diff
@@ -39,7 +39,35 @@ public override void Populate(TextWriter trapFile)
             trapFile.compilation_assembly(this, assembly);
 
             // Arguments
-            Compilation.Settings.Args.ForEach((arg, index) => trapFile.compilation_args(this, index, arg));
+            var expandedIndex = 0;
+            for (var i = 0; i < Compilation.Settings.Args.Length; i++)
+            {
+                var arg = Compilation.Settings.Args[i];
+                trapFile.compilation_args(this, i, arg);
+
+                if (CommandLineExtensions.IsFileArgument(arg))
+                {
+                    try
+                    {
+                        var rspFileContent = System.IO.File.ReadAllText(arg[1..]);
+                        var rspArgs = CommandLineParser.SplitCommandLineIntoArguments(rspFileContent, removeHashComments: true);
+                        foreach (var rspArg in rspArgs)
+                        {
+                            trapFile.compilation_expanded_args(this, expandedIndex, rspArg);
+                            expandedIndex++;
```

Comment:
I think we tend to write the shorter version elsewhere in the codebase:

```suggestion
                            trapFile.compilation_expanded_args(this, expandedIndex++, rspArg);
```